static/js/components/ValidationAlert.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import _ from 'lodash';
+
+import type { ValidationErrors } from '../flow/profileTypes';
+
+export default class ValidationAlert extends React.Component {
+  props: {
+    errors:   ValidationErrors,
+    children: React$Element[],
+  };
+
+  alertMessage: Function = (): React$Element|void => {
+    const { errors } = this.props;
+    if ( !_.isEmpty(errors) ) {
+      return (
+        <span className="message">
+          Please fix the errors above and try again
+        </span>
+      );
+    }
+  };
+
+  render () {
+    const { children } = this.props;
+    return (
+      <div className="validation-alert">
+        { this.alertMessage() }
+        <div className="actions">
+          { children }
+        </div>
+      </div>
```

Comment:
3 main problems i still have

1. i don't think it makes semantic sense for us to pass `Button`s or any other component as children to a component called `ValidationAlert`. _semantically_ i don't know why i would be wrapping `Button`s in that component
1. this still outputs an unnecessary `<div className="validation-alert">` wrapper element when there are no errors in a Dialog
1. not terribly important, but it's easier to reason about the error message component and the general component tree if it's not invoking `{children}`

**with that said**, i'm getting the sense that we're at an impasse, and this isn't going to introduce some kind of show-stopping tech debt. you're good with your approach if i haven't convinced you 😄